client/src/components/TODOList/TODOListViewer.js

```diff
@@ -17,38 +18,28 @@ const TODOListViewer = ({ selectedCategory }) => {
   // updated
   useEffect(() => {
     setLoading(true);
-    dispatch(thunk.getTODOListAsync()).then(() => {
+    dispatch(
+      thunk.getTODOListAsync({
+        userID: user.user.userID,
+        categoryID: selectedCategoryID,
+      })
+    ).then(() => {
       setLoading(false);
     });
-  }, [dispatch, fetchTODOList]);
-
-  // isVisibleTODOListEmpty checks if the todos array is empty.
-  const isVisibleTODOListEmpty = () => {
-    return getVisibleTODOs().length === 0;
-  };
-
-  // The getVisibleTODOs function is defined to filter the TODOList based on the selected category.
-  const getVisibleTODOs = () => {
-    if (!selectedCategory) {
-      return TODOList;
-    }
-    return TODOList.filter((todo) => todo.category === selectedCategory);
-  };
+  }, [dispatch, fetchTODOList, selectedCategoryID, user.user.userID]);
 
   // Only render the ListView if the isVisibleTODOListEmpty function returns false,
   // indicating that there are visible TODOItems to display.
   return (
     <div>
-      <header className="flex justify-between">
+      <header className="flex justify-between flex-wrap">
         <Typography variant="h5">TODO List</Typography>
-        <Typography variant="small">
-          Total Todos: {getVisibleTODOs().length}
+        <Typography variant="small" className="">
```

Comment:
![image](https://github.com/awctw/study-dash/assets/56493743/6a4ab0ab-9752-4e44-980d-403a359e36d2)

No spacing between "TODO List" and Total Todos where there was before. Also unused className.

Reply:
I will fix this